common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOtherCache.java

```diff
@@ -0,0 +1,68 @@
+package com.microsoft.identity.common.internal.cache;
+
+import android.content.Context;
+import androidx.annotation.VisibleForTesting;
+import com.microsoft.identity.common.adal.internal.cache.IStorageHelper;
+import com.microsoft.identity.common.adal.internal.cache.StorageHelper;
+
+public class MsalOtherCache {
+
+    /**
+     * The name of the SharedPreferences file on disk.
+     * This file is only used to store MSAL CPP authority validation metadata
+     */
+    public static final String DEFAULT_CPP_AUTHORITY_VALIDATION_METADATA_SHARED_PREFERENCES =
+            "com.microsoft.identity.client.cpp_authority_validation_metadata";
+
+    private final ICacheKeyValueDelegate mCacheKeyValueDelegate;
+
+    private final ISharedPreferencesFileManager mCppAuthorityValidationMetadataSharedPreferencesFileManager;
+
+    public MsalOtherCache(final Context context)
+    {
+        mCacheKeyValueDelegate = new CacheKeyValueDelegate();
+
+        final IStorageHelper storageHelper = new StorageHelper(context);
+        mCppAuthorityValidationMetadataSharedPreferencesFileManager =
+                SharedPreferencesFileManager.getSharedPreferences(
+                        context,
+                        DEFAULT_CPP_AUTHORITY_VALIDATION_METADATA_SHARED_PREFERENCES,
+                        storageHelper

```

Comment:
`null` here, as this data doesn't need to be encrypted (and would become data that would need to be migrated later is encryption keys are rotated)

Reply:
The same as above